internal/pkg/service/stream/cleanup/node.go

```diff
@@ -1,115 +0,0 @@
-package cleanup
-
-import (
-	"context"
-	"strings"
-	"time"
-
-	"github.com/benbjohnson/clock"
-	etcd "go.etcd.io/etcd/client/v3"
-	"golang.org/x/sync/semaphore"
-
-	"github.com/keboola/keboola-as-code/internal/pkg/log"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/distribution"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/task"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
-	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
-)
-
-const (
-	// FileExpirationDays defines how old files are to be deleted.
-	FileExpirationDays = 1
-	// MaxTasksPerNode limits number of parallel cleanup tasks per node.
-	MaxTasksPerNode         = 20
-	ReceiverCleanupTimeout  = 5 * time.Minute
-	taskTypeReceiverCleanup = "receiver.cleanup"
-)
-
-type dependencies interface {
-	Clock() clock.Clock
-	Logger() log.Logger
-	Schema() *schema.Schema
-	EtcdClient() *etcd.Client
-	Store() *store.Store
-	DistributionNode() *distribution.Node
-	TaskNode() *task.Node
-	StatisticsRepository() *statistics.Repository
-}
-
-type Node struct {
-	deps   dependencies
-	client *etcd.Client
-	logger log.Logger
-	schema *schema.Schema
-	dist   *distribution.Node
-	tasks  *task.Node
-	sem    *semaphore.Weighted
-}
-
-func NewNode(d dependencies, logger log.Logger) *Node {
-	return &Node{
-		deps:   d,
-		client: d.EtcdClient(),
-		logger: logger,
-		schema: d.Schema(),
-		dist:   d.DistributionNode(),
-		tasks:  d.TaskNode(),
-		sem:    semaphore.NewWeighted(MaxTasksPerNode),
-	}
-}
-
-func (n *Node) Check(ctx context.Context) error {
-	tasksCount := 0
-	err := n.schema.
-		Configs().
-		Receivers().
-		GetAll().
-		Do(ctx, n.client).
-		ForEachValue(func(v model.ReceiverBase, header *iterator.Header) error {
```

Comment:
Removed old cleanup implementation, it was iterating `receivers/sources` and a task was created for each source.

Not all files are iterated directly (there is an interator), so it should be faster and simpler.